components/properties/src/trievalue.rs

```diff
@@ -163,3 +173,22 @@ impl TrieValue for GeneralCategoryGroup {
         u16::try_from(i).map(packed_u16_to_gcg)
     }
 }
+
+impl TrieValue for MirroredPairedBracketData {
+    type TryFromU32Error = CharTryFromError;
+
+    fn try_from_u32(i: u32) -> Result<Self, Self::TryFromU32Error> {
```

Comment:
Nit/Suggestion: you already have `impl TryFrom<u32> for MirroredPairedBracketData` and you should call it from here.

Reply:
I think what you did here is good; map to `ZeroVecError` when you need to, and otherwise stick with a more expressive error type.

I observe that this function allows all values of the open enum; `BidiPairedBracketType` takes 0, 1, 2, or 3. I have another thread for discussing whether that is good or bad or neutral.